pkg/band/rpc.go

```diff
@@ -58,3 +59,138 @@ func GetPhyVersions(ctx context.Context, req *ttnpb.GetPhyVersionsRequest) (*ttn
 	}
 	return res, nil
 }
+
+// ListBands returns the list of supported bands.
+func ListBands(ctx context.Context, req *ttnpb.ListBandsRequest) (*ttnpb.ListBandsResponse, error) {
+	filteredVersions := make(map[string]map[ttnpb.PHYVersion]Band)
+
+	if req.BandId != "" {
+		versions, ok := All[req.BandId]
+		if !ok {
+			return nil, errBandNotFound.WithAttributes(req.BandId, req.PhyVersion.String())
+		}
+
+		filteredVersions[req.BandId] = versions
+	}
+
+	if len(filteredVersions) == 0 {
+		filteredVersions = All
```

Comment:
Clone the map using `maps.Clone`, otherwise the `delete` and the assignment below will edit the global variable.

Reply:
Did not know about this behaviour, fixed it